### PR TITLE
[Fix](nereids) modify the binding aggregate function in order by

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
modify the bind logical to make the order by has same behavior with mysql when sort child is aggregate.
when an order by  Expr has aggregate function, all slots in this order by Expr should bind the LogicalAggregate non-AggFunction outputs  first, then bind the LogicalAggregate Child
e.g.
select 2*abs(sum(c1)) as c1, c1,sum(c1)+c1 from t_order_by_bind_priority group by c1 order by sum(c1)+c1 asc;
in this sql, the two c1 in order by all bind to the c1 in  t_order_by_bind_priority

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by morrySnow** [comment]
add description

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java [hunk 1]
```diff
@@ -763,9 +763,38 @@ private Plan bindSortWithoutSetOperation(MatchingContext<LogicalSort<Plan>> ctx)
...
         Builder<OrderKey> boundOrderKeys = ImmutableList.builderWithExpectedSize(sort.getOrderKeys().size());
+        FunctionRegistry functionRegistry = cascadesContext.getConnectContext().getEnv().getFunctionRegistry();
         for (OrderKey orderKey : sort.getOrderKeys()) {
-            Expression boundKey = analyzer.analyze(orderKey.getExpr());
+            Expression boundKey;
+            if (hasAggregateFunction(orderKey.getExpr(), functionRegistry)) {
+                boundKey = bindInInputChildScope.analyze(orderKey.getExpr());
+            } else {
+                boundKey = bindInInputScopeThenInputChildScope.analyze(orderKey.getExpr());
+            }
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java:797]: we should bind bindInInputChildScope when **in** aggregate function, not equals to **has** aggregate function,
see `bindHavingAggregate`, there has a visitor to do this function

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by 924060929 — DISMISSED** [review]

**Review by starocean999 — DISMISSED** [review]

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by 924060929 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**